src/anomalib/utils/visualization/image.py

```diff
@@ -136,38 +138,38 @@ def _visualize_batch(self, batch: dict) -> Iterator[GeneratorResult]:
-            if "image_path" in batch:
-                height, width = batch["image"].shape[-2:]
-                image = (read_image(path=batch["image_path"][i]) * 255).astype(np.uint8)
+            if isinstance(batch, ImageBatch):
+                height, width = batch.image.shape[-2:]
+                image = (read_image(path=batch.image_path[i]) * 255).astype(np.uint8)
                 image = cv2.resize(image, dsize=(width, height), interpolation=cv2.INTER_AREA)
-            elif "video_path" in batch:
-                height, width = batch["image"].shape[-2:]
-                image = batch["original_image"][i].squeeze().cpu().numpy()
+            elif isinstance(batch, VideoBatch):
+                height, width = batch.image.shape[-2:]
+                image = batch.original_image[i].squeeze().cpu().permute(1, 2, 0).numpy()
                 image = cv2.resize(image, dsize=(width, height), interpolation=cv2.INTER_AREA)
             else:
-                msg = "Batch must have either 'image_path' or 'video_path' defined."
-                raise KeyError(msg)
+                msg = "Batch must be of type ImageBatch or VideoBatch."
+                raise TypeError(msg)
 
             file_name = None
-            if "image_path" in batch:
-                file_name = Path(batch["image_path"][i])
-            elif "video_path" in batch:
-                zero_fill = int(np.log10(batch["last_frame"][i])) + 1
-                suffix = f"{str(batch['frames'][i].int().item()).zfill(zero_fill)}.png"
-                file_name = Path(batch["video_path"][i]) / suffix
+            if isinstance(batch, ImageBatch):
+                file_name = Path(batch.image_path[i])
+            elif isinstance(batch, VideoBatch):
+                zero_fill = int(np.log10(batch.last_frame[i].cpu())) + 1
+                suffix = f"{str(batch.frames[i].int().item()).zfill(zero_fill)}.png"
+                file_name = Path(batch.video_path[i]) / suffix
 
             image_result = ImageResult(
                 image=image,
-                pred_score=batch["pred_scores"][i].cpu().numpy().item() if "pred_scores" in batch else None,
-                pred_label=batch["pred_labels"][i].cpu().numpy().item() if "pred_labels" in batch else None,
-                anomaly_map=batch["anomaly_maps"][i].cpu().numpy() if "anomaly_maps" in batch else None,
-                pred_mask=batch["pred_masks"][i].squeeze().int().cpu().numpy() if "pred_masks" in batch else None,
-                gt_mask=batch["mask"][i].squeeze().int().cpu().numpy() if "mask" in batch else None,
-                gt_boxes=batch["boxes"][i].cpu().numpy() if "boxes" in batch else None,
-                pred_boxes=batch["pred_boxes"][i].cpu().numpy() if "pred_boxes" in batch else None,
-                box_labels=batch["box_labels"][i].cpu().numpy() if "box_labels" in batch else None,
+                pred_score=batch.pred_score[i].cpu().numpy().item() if batch.pred_score is not None else None,
```

Comment:
Yes, we will get rid of the `ImageResult`, but this will be a part of the wider visualization refactor. The current `NumpyImageItem` -> `ImageResult` conversion is temporary to make our old visualizer compatible with the new dataclasses design. Once we update the visualizer we will remove `ImageResult`.